File: src/Moryx.ControlSystem.ProcessEngine.Web/src/app/api/models/Moryx/ControlSystem/Jobs/Endpoints/job-model.ts
/* tslint:disable */
/* eslint-disable */
import { ProductionJobModel as MoryxControlSystemJobsEndpointsProductionJobModel } from '../../../../../models/Moryx/ControlSystem/Jobs/Endpoints/production-job-model';
import { SetupJobModel as MoryxControlSystemJobsEndpointsSetupJobModel } from '../../../../../models/Moryx/ControlSystem/Jobs/Endpoints/setup-job-model';
export interface JobModel {
  canAbort?: boolean;
  canComplete?: boolean;
  displayState?: string | null;
  id?: number;
  isRunning?: boolean;
  isWaiting?: boolean;
  productionJob?: MoryxControlSystemJobsEndpointsProductionJobModel;
  recipeId?: number;
  setupJob?: MoryxControlSystemJobsEndpointsSetupJobModel;
  state?: string | null;
  workplan?: string | null;
}

File: src/Moryx.ControlSystem.Jobs.Endpoints/Models/JobModel.cs
// Copyright (c) 2025, Phoenix Contact GmbH & Co. KG
// Licensed under the Apache License, Version 2.0

using System.Runtime.Serialization;

namespace Moryx.ControlSystem.Jobs.Endpoints
{
    [DataContract]
    public class JobModel
    {
        [DataMember]
        public long Id { get; set; }

        [DataMember]
        public long RecipeId { get; set; }

        [DataMember]
        public string State { get; set; }

        [DataMember]
        public string DisplayState { get; set; }

        [DataMember]
        public string Workplan { get; set; }

        [DataMember]
        public bool CanComplete { get; set; }

        [DataMember]
        public bool CanAbort { get; set; }

        [DataMember]
        public bool IsWaiting { get; set; }

        [DataMember]
        public bool IsRunning { get; set; }

        [DataMember]
        public ProductionJobModel ProductionJob { get; set; }

        [DataMember]
        public SetupJobModel SetupJob { get; set; }
    }
}

